var/spack/repos/builtin/packages/hpctoolkit/package.py

```diff
@@ -64,33 +64,29 @@ class Hpctoolkit(AutotoolsPackage):
         "the hardware performance counters.",
     )
 
-    variant(
-        "all-static",
-        default=False,
-        description="Needed when MPICXX builds static binaries for the compute nodes.",
-    )
-
-    variant(
-        "cuda", default=False, description="Support CUDA on NVIDIA GPUs (2020.03.01 or later)."
-    )
+    # Accelerator variants: cuda, rocm, etc.
+    variant("cuda", default=False, description="Support CUDA on NVIDIA GPUs.", when="@2020.03:")
 
     variant(
         "level_zero",
         default=False,
-        description="Support Level Zero on Intel GPUs (2022.04.15 or later).",
+        description="Support Level Zero on Intel GPUs.",
+        when="@2022.04:",
     )
 
     variant(
         "gtpin",
         default=False,
         description="Support instrumenting Intel GPU kernels with Intel GT-Pin "
-        "(2022.05.15 or later, and requires level_zero).",
+        "(requires level_zero).",
+        when="@2022.05:",
```

Comment:
Not sure if I understand the description correctly, but in case it makes sense you can enforce:
```suggestion
        "(requires level_zero).",
        when="@2022.05: +level_zero",
```

Reply:
Oh, I didn't realize that you could make conditional variants depend
on other variants.   If that's legal spack, then yes it has the correct semantics.
Let me test it and I'll update the PR.